automation/src/test/java/com/microsoft/identity/common/test/automation/tasks/AcquireTokenCloud.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.identity.common.test.automation.tasks;
+
+import com.microsoft.identity.common.test.automation.actors.User;
+import com.microsoft.identity.common.test.automation.interactions.CloseKeyboard;
+import com.microsoft.identity.common.test.automation.ui.Main;
+import com.microsoft.identity.common.test.automation.ui.Request;
+import com.microsoft.identity.common.test.automation.ui.identityproviders.AADV1.SignInPagePassword;
+import com.microsoft.identity.common.test.automation.ui.identityproviders.AADV1.SignInPageUserName;
+import com.microsoft.identity.common.test.automation.utility.TokenRequest;
+
+import net.serenitybdd.screenplay.Actor;
+import net.serenitybdd.screenplay.Task;
+import net.serenitybdd.screenplay.actions.Click;
+import net.serenitybdd.screenplay.actions.Enter;
+import net.serenitybdd.screenplay.waits.WaitUntil;
+import net.thucydides.core.annotations.Steps;
+
+import static net.serenitybdd.screenplay.matchers.WebElementStateMatchers.isVisible;
+
+public class AcquireTokenCloud implements Task{
+
+    @Steps
+    CloseKeyboard closeKeyboard;
+
+    String prompt;
+    String userIdentifier;
+
+    @Override
+    public <T extends Actor> void performAs(T actor) {
+        User user = (User)actor;
+        TokenRequest tokenRequest = user.getTokenRequest();
+        tokenRequest.setPromptBehavior(prompt);
+        tokenRequest.setUserIdentitfier(userIdentifier);
+        SignInUser signInUser = SignInUser.GetSignInUserByFederationProvider(user.getFederationProvider());
+        actor.attemptsTo(
+                WaitUntil.the(Main.ACQUIRE_TOKEN_BUTTON, isVisible()).forNoMoreThan(10).seconds(),
+                Click.on(Main.ACQUIRE_TOKEN_BUTTON),
+                Enter.theValue(user.getTokenRequestAsJson()).into(Request.REQUEST_INFO_FIELD),
+                closeKeyboard,
+                WaitUntil.the(Request.SUBMIT_REQUEST_BUTTON, isVisible()).forNoMoreThan(10).seconds(),
+                Click.on(Request.SUBMIT_REQUEST_BUTTON),
+                WaitUntil.the(SignInPageUserName.USERNAME, isVisible()).forNoMoreThan(10).seconds(),
+                new EnterUserNameForSignInDisambiguation(),
+                WaitUntil.the(SignInPagePassword.PASSWORD, isVisible()).forNoMoreThan(10).seconds(),
+                signInUser
```

Comment:
The test is failing on this step on my device. Can you verify if its working on your physical device?
You need to add "SignInUserCloud" to the SignInUser task in the switch case. Currently it's trying to do the default SignInUserADFSv2 and its fails due to that